### PR TITLE
LibGL: Implement missing features and bug fixes for glquake

### PR DESCRIPTION
**Pull request by sunverwerth**
#7062 

Implement missing features needed by glquake.
Also fixes several bugs that occured while porting.

There are still some functions missing:

* glTexEnvf
* glPixelStorei
* glTexSubImage2D
* glDrawBuffer
* glPolygonMode
* glPolygonOffset
* glFogi
* glFogf
* glFogfv

### CODE REVIEW

**Review by Quaker762 — APPROVED** [review]
Looks good to me!

File: Userland/Libraries/LibGL/Tex/Texture2D.cpp [hunk 1]
```diff
@@ -18,11 +18,13 @@ void Texture2D::upload_texture_data(GLenum, GLint lod, GLint internal_format, GL
...
-    // Apparently this allocates memory on the GPU (according to Khronos docs..)?
+    // Somebody passed us in nullptr... allocate uninitialized memory and set size
     if (pixels == nullptr) {
-        dbgln("LibGL: pixels == nullptr when uploading texture data.");
-        VERIFY_NOT_REACHED();
+        auto& mip = m_mipmaps[lod];
+        mip.set_width(width);
+        mip.set_height(height);
+        mip.pixel_data().resize(width * height);
+        return;
```
Comment (alimpfard) [inline Userland/Libraries/LibGL/Tex/Texture2D.cpp:27]: This doesn't seem related to the commit it's in.

**Comment by awesomekling** [comment]
This has conflicts.

File: Userland/Libraries/LibGL/Tex/Texture2D.cpp (hunk 1 above)
Comment (sunverwerth) [reply Userland/Libraries/LibGL/Tex/Texture2D.cpp:27]: Yes, I'll put it into a separate one

**Comment by sunverwerth** [comment]
Rebased, fixed conflicts and split one commit. @alimpfard @awesomekling

**Comment by awesomekling** [comment]
Please wrap commit messages at 72 characters, per the [contributing guidelines](https://github.com/SerenityOS/serenity/blob/6c9343e262abe5a72f8b2f0d71590c627a7eb622/CONTRIBUTING.md)

**Comment by sunverwerth** [comment]
> 
> 
> Please wrap commit messages at 72 characters, per the [contributing guidelines](https://github.com/SerenityOS/serenity/blob/6c9343e262abe5a72f8b2f0d71590c627a7eb622/CONTRIBUTING.md)

@awesomekling Which commit do you mean? The longest line I see is "The rasterizer always expects triangles with a consistent winding." at 66 characters. In https://github.com/SerenityOS/serenity/pull/9451/commits/39facfd66afb462faac75aa0421e5e4e54298f99
